UnitySDK/Assets/ML-Agents/Scripts/Grpc/RpcCommunicator.cs

```diff
@@ -424,6 +425,53 @@ private void HandleOnPlayModeChanged(PlayModeStateChange state)
             }
         }
 
+        private void CacheBrainParameters(string brainKey, BrainParameters brainParameters)
+        {
+            if (m_sentBrainKeys.Contains(brainKey))
+            {
+                return;
```

Comment:
I would almost want to compare the brainParameters here to the cached ones to ensure they are the same (with assert). This might be too much work for something that will change soon.

Reply:
Sure, that's reasonable.

Reply:
Punting on this for now